### PR TITLE
Collapsible test panel

### PR DESCRIPTION
**Pull request by bautistaaa**
closes https://github.com/bautistaaa/typehero/issues/511

needs the @PickleNik touch

### CODE REVIEW

File: apps/web/src/app/challenge/_components/settings/settings-form.tsx [hunk 1]
```diff
@@ -41,6 +41,8 @@ export function SettingsForm() {
     defaultValues: settings,
   });
 
+  console.log(form.formState.errors);
```
Comment (nartc) [inline apps/web/src/app/challenge/_components/settings/settings-form.tsx:44]: 🤔

File: packages/monaco/src/index.tsx [hunk 2]
```diff
@@ -90,87 +86,122 @@ export function CodePanel(props: CodePanelProps) {
...
+        monaco={monacoInstance}
+        expandTestPanel={isTestPanelExpanded}
+        tests={props.challenge.tests}
+        userCode={code}
+        onMount={{
+          tests: async (editor, monaco) => {
+            const model = editor.getModel();
+
+            if (!model) {
+              throw new Error();
```
Comment (nartc) [inline packages/monaco/src/index.tsx:107]: question: what is handling this error? If it's some ErrorBoundary, should we specify the error message?

File: packages/monaco/src/index.tsx [hunk 3]
```diff
@@ -90,87 +86,122 @@ export function CodePanel(props: CodePanelProps) {
...
+            size="sm"
+            onClick={() => {
+              setIsTestPanelExpanded((tp) => !tp);
+            }}
+          >
+            Tests
+            {isTestPanelExpanded ? <ChevronDown size={16} /> : <ChevronUp size={16} />}
+          </Button>
+          {hasFailingTest && <XCircle className="stroke-red-600 dark:stroke-red-300" />}
+          {!hasFailingTest && <CheckCircle2 className="stroke-green-600 dark:stroke-green-300" />}
```
Comment (nartc) [inline packages/monaco/src/index.tsx:202]: nitpick: ternary for `isTestPanelExpanded` but not for `hasFailingTest`?

**Review by nartc — COMMENTED** [review]
I just want to heckle Trash

File: apps/web/src/app/challenge/_components/settings/settings-form.tsx (hunk 1 above)
Comment (bautistaaa) [reply apps/web/src/app/challenge/_components/settings/settings-form.tsx:44]: WIP MAN!

File: packages/monaco/src/index.tsx (hunk 2 above)
Comment (bautistaaa) [reply packages/monaco/src/index.tsx:107]: not sure need to check. iirc the editor just stops working and it probably goes to sentry will confirm later!

File: packages/monaco/src/index.tsx (hunk 3 above)
Comment (bautistaaa) [reply packages/monaco/src/index.tsx:202]: yep yep ill fix

File: apps/web/src/app/challenge/_components/settings/settings-form.tsx (hunk 1 above)
Comment (nartc) [reply apps/web/src/app/challenge/_components/settings/settings-form.tsx:44]: I didn't see "Draft"

Comment (bautistaaa) [reply apps/web/src/app/challenge/_components/settings/settings-form.tsx:44]: never made a draft PR in my life

File: packages/monaco/src/index.tsx (hunk 3 above)
Comment (maxdemaio) [reply packages/monaco/src/index.tsx]: Solved!

File: apps/web/src/app/challenge/_components/settings/settings-form.tsx (hunk 1 above)
Comment (maxdemaio) [reply apps/web/src/app/challenge/_components/settings/settings-form.tsx]: All set 😤

File: packages/email/yarn.lock
Comment (jhechtf) [inline packages/email/yarn.lock:1]: why is the email package using yarn???

Comment (Hacksore) [reply packages/email/yarn.lock:1]: https://github.com/resendlabs/react-email

**Review by jhechtf — APPROVED** [review]
LGTM but definitely needs @PickleNik 's touch. I do wonder about the stray yarn.lock file @bautistaaa

File: packages/email/yarn.lock
Comment (Hacksore) [reply packages/email/yarn.lock:1]: https://github.com/resendlabs/react-email/pull/892

Comment (jhechtf) [reply packages/email/yarn.lock:1]: I see; that's such a weird thing to program into their code but glad you got it resolved.

**Pull request merged**